### PR TITLE
Remove legacy katz

### PR DESCRIPTION
**Pull request by ChuckHastings**
Remove the legacy Katz implementation which is no longer used.

Marked as breaking since it deletes a C++ API.

### CODE REVIEW

**Review by seunghwak — APPROVED** [review]
Approving the C++ part.

File: conda/environments/cugraph_dev_cuda11.5.yml [hunk 1]
```diff
@@ -47,3 +49,6 @@ dependencies:
 - pytest-cov
 - gtest=1.10.0
 - gmock=1.10.0
+- ptxcompiler  # [linux64]
+- gcc_linux-64=9.* # [linux64]
+- sysroot_linux-64==2.17 # [linux64]
```
[the remark this answers is not in the document]
Comment (ChuckHastings) [reply conda/environments/cugraph_dev_cuda11.5.yml:54]: Good point.  I actually accidentally included this in the PR, this update (along with the other files) should probably be in a separate PR.  Let me see if I can do that.

[the remark this answers is not in the document]
Comment (ChuckHastings) [reply conda/environments/cugraph_dev_cuda11.5.yml]: Moved this change into a separate PR that addresses all of the CUDA versions.

[the remark this answers is not in the document]
Comment (ChuckHastings) [reply conda/environments/cugraph_dev_cuda11.5.yml]: Moved this change into a separate PR that addresses all of the CUDA versions.

**Review by rlratzel — APPROVED** [review]

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**